guardrails/utils/pydantic_utils/v1.py

```diff
@@ -238,12 +251,30 @@ def convert_pydantic_model_to_openai_fn(model: BaseModel) -> Dict:
         OpenAI function paramters.
     """
 
+    schema_model = model
+
+    type_origin = get_origin(model)
+    if type_origin == list:
+        item_types = get_args(model)
+        if len(item_types) > 1:
+            raise ValueError("List data type must have exactly one child.")
+        # No List[List] support; we've already declared that in our types
+        schema_model = safe_get(item_types, 0)
+
     # Create a bare model with no extra fields
     class BareModel(BaseModel):
-        __annotations__ = model.__annotations__
+        __annotations__ = schema_model.__annotations__
```

Comment:
nit: you can use `_create_bare_model` here

Reply:
Done